docker/pypi/wmagent/init.sh

```diff
@@ -329,12 +331,32 @@ EOF
     echo "-----------------------------------------------------------------------"
 }
 
-check_docker_init() {
-    # A function to check all previously populated */.dockerInit files
-    # from all previous steps and compare them with the /data/.dockerBuildId
+check_wmagent_init() {
+    # A function to check all previously populated */.init<step> files
+    # from all previous steps and compare them with the /data/.wmaBuildId
     # if all do not match we cannot continue - we consider configuration/version
     # mismatch between the host and the container
 
+    # NOTE: On every step we need to check the .init<step> file content. We always compare
+    #       the current $WMA_BUILD_ID with the one previously initialized at the host.
+    #       The WMA_BUILD_ID is generated during the execution of `install.sh`
+    #       at build time (for docker images) or deploy time for virtual env
+    #       There are few levels of comparison we can make:
+    #       * if we want to trigger re-initialization on any WMAgent image rebuild,
+    #         then the $WMA_BUILD_ID should contain a sha256sum of a random variable
+    #       * if we want to trigger re-initialization only on new WMAgent tag builds,
+    #         then the $WMA_BUILD_ID should contain a sha256sum of whole $WMA_TAG
+    #       * if we want to trigger re-initialization only on version change (not on patches),
+    #         then we should split $WMA_TAG in parts: major, minor and patch(release candidate)
+    #         part and the $WMA_BUILD_ID should contain a sha256sum only of the major + minor
+    #         part excluding the patch version, e.g.:
+    #         WMA_TAG=2.3.3.1;
+    #         WMA_VER[release]=2.3.3
+    #         WMA_VER[major]=2.3
+    #         WMA_VER[minor]=3
+    #         WMA_VER[patch]=1
+    #       The current implementation considers the first one - re-initialization on any WMAgent rebuild
```

Comment:
This last line in this comment is also confusing to me:
"The current implementation considers the first one - re-initialization on any WMAgent rebuild"

What is "current implementation"? What is already in the repository? Or what is provided in this PR?

In addition, I wonder in practice what is the actual meaning of "re-initialization on any WMAgent rebuild"? Is it that when we run a new container version, we will perform every single step of the initialization again? E.g.:
1) defining cronjobs
2) setup couchapps
3) (re-)create the agent config.py
4) create the SQL database schema
5) and so on?

Reply:
No, it meant that whenever somebody goes and executes `install.sh` which is a step during the docker image build process for the production agents and an equivalent step during the creation of the virtual env for the agents who are about to use it, the resultant docker image or virtual env root would have their own unique to them id and they must go through the initialization process at least once. But not on every restart. 
The current implementation is not this one any more. We switched to the model where we restart the initialization process only whenever someone changes  the release version.